### PR TITLE
multiple warning fixes and build updates:

### PR DESCRIPTION
**Pull request by sezero**
- src/*.[ch]: multiple fixes for gcc -Wmisleading-indentation, -Wshadow,
  -Wsign-compare, -Wset-but-unused, etc warnings. Fixes for Watcom W007
  warnings. (Most from https://github.com/Konstanty/libmodplug/issues/63
  are fixed, although I might have missed a few. The -Wodr warnings are
  not fixed.)

- src/*.[ch]: build fixes for older MSVC versions.

- configure.ac: revert the hardcoded -macosx-version-min switches from
  commit 28f938cfa5d. it was a very bad idea.

- configure.ac: does not really require ac-2.71: work fine down to 2.63.

- configure.ac: remove unnecessary stuff.

- configure.ac: update libm need and discovery. update src/Makefile.am
  accordingly.

- cmake updates to make it more usable:
  - fix header list after commit 664512e0d
  - add -Wall for gcc and clang
  - add --no-undefined linker flag if supported
  - make -lm conditional
  - add support for visibility attributes
  - set VERSION / SOVERSION properties of unix shared lib
  - set dylib version numbers to match libtoolized builds
  - add test for WORDS_BIGENDIAN
  - rearrange the file

- Makefile.am: add CMakeLists.txt to EXTRA_DIST

### CODE REVIEW

File: src/load_it.cpp [hunk 1]
```diff
@@ -237,9 +238,8 @@ BOOL CSoundFile::ReadIT(const BYTE *lpStream, DWORD dwMemLength)
 	if (inspossize > MAX_INSTRUMENTS) inspossize = MAX_INSTRUMENTS;
 	inspossize <<= 2;
 	memcpy(inspos, lpStream+dwMemPos, inspossize);
-	for (UINT j=0; j < (inspossize>>2); j++)
-	{
-	       inspos[j] = bswapLE32(inspos[j]);
+	for (j=0; j < (inspossize>>2); j++) {
+	    inspos[j] = bswapLE32(inspos[j]);
```
Comment (AliceLR) [inline src/load_it.cpp:242]: The loop style changes don't really match anything else in the file.

Most of the other source file changes look reasonable (can't really speak for the configury changes).

Comment (sezero) [reply src/load_it.cpp:242]: Hmph, the loop style changes crept in from my fork which were accompanied by build/warning fixes.
I can take them back if @Konstanty insists on it.

**Comment by sezero** [comment]
Rebased.

**Comment by sezero** [comment]
The change reverted by https://github.com/Konstanty/libmodplug/commit/500711c281ce35d90504684f87ce5c1384cb95bb was actually meant to fix build using old MSVC versions: It would be nice to have it back, but it's not end of the world either.

**Comment by Konstanty** [comment]
Is the issue for old MSVC reusing a variable in two different for loops in the same scope? or just not declaring the variable at the start of the function?

**Comment by sezero** [comment]
> Is the issue for old MSVC reusing a variable in two different for loops in the same scope?

Yes, it barfs with redefinition errors

**Comment by sezero** [comment]
> > Is the issue for old MSVC reusing a variable in two different for loops in the same scope?
> 
> Yes, it barfs with redefinition errors

FWIW, here are the errors:
```
sndfile.cpp(1406) : error C2374: 'j' : redefinition; multiple initialization
        sndfile.cpp(1400) : see declaration of 'j'

load_it.cpp
load_it.cpp(244) : error C2374: 'j' : redefinition; multiple initialization
        load_it.cpp(234) : see declaration of 'j'
load_it.cpp(254) : error C2374: 'j' : redefinition; multiple initialization
        load_it.cpp(234) : see declaration of 'j'
```

Re-applying the changes makes it build properly.

**Pull request merged**